core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorter.java

```diff
@@ -143,6 +151,8 @@ private UnsafeExternalSorter(
       this.inMemSorter = existingInMemorySorter;
     }
     this.peakMemoryUsedBytes = getMemoryUsage();
+    this.numElementsForSpillThreshold =
+      SparkEnv.get().conf().getLong("spark.shuffle.spill.numElementsForceSpillThreshold", (1024 * 1024 * 1024 / 8));
```

Comment:
numElementsForSpillThreshold should be not greater than 1024 * 1024 * 1024 / 4, the element require 16 bytes.